tests/functional/test_snap_dcgm.py

```diff
@@ -0,0 +1,65 @@
+import json
+import subprocess
+import time
+
+
+def test_dcgm_exporter_endpoint():
+    """Test of the dcgm-exporter service and its endpoint."""
+    endpoint = "localhost:9400/metrics"
+    timeout = 60  # seconds
+    start_time = time.time()
+
+    def query_endpoint():
+        return subprocess.run(["curl", endpoint], text=True, capture_output=True)
+
+    while (result := query_endpoint()).returncode != 0 or not result.stdout.strip():
+        if time.time() - start_time > timeout:
```

Comment:
nit: This packs a lot of logic in the `while` condition and is a little hard to read - could we go with a method that doesn't involve declaring a nested function or having so much happening in the condition here?

Reply:
I kind agree... what if we use something like `tenacity` where we specify the number of runs, time and etc?

Reply:
@Deezzir please don't resolve conversations on reviews unless it's very clear that they've been addressed.  General etiquette on our team is to let it be resolved by the person who started the conversation. :)

Reply:
Hmm actually it looks like several have been addressed, but github is not making that clear at all... :(

But in general please let the author of the thread resolve it; it makes it more confusing otherwise.